sources/models/src/modeled_types/oci_defaults.rs

```diff
@@ -120,10 +157,158 @@ mod oci_defaults_rlimits {
 
     #[test]
     fn linux_rlimit_strings() {
-        check_rlimit_strings(
-            OciDefaultsResourceLimitType::MaxOpenFiles,
-            "max-open-files",
-            "RLIMIT_NOFILE",
-        );
+        let rlimits = [
+            (MaxAddressSpace, "max-address-space", "RLIMIT_AS"),
+            (MaxCoreFileSize, "max-core-file-size", "RLIMIT_CORE"),
+            (MaxCpuTime, "max-cpu-time", "RLIMIT_CPU"),
+            (MaxDataSize, "max-data-size", "RLIMIT_DATA"),
+            (MaxFileLocks, "max-file-locks", "RLIMIT_LOCKS"),
+            (MaxFileSize, "max-file-size", "RLIMIT_FSIZE"),
+            (MaxLockedMemory, "max-locked-memory", "RLIMIT_MEMLOCK"),
+            (MaxMsgqueueSize, "max-msgqueue-size", "RLIMIT_MSGQUEUE"),
+            (MaxNicePriority, "max-nice-priority", "RLIMIT_NICE"),
+            (MaxOpenFiles, "max-open-files", "RLIMIT_NOFILE"),
+            (
+                MaxPendingSignals,
+                "max-pending-signals",
+                "RLIMIT_SIGPENDING",
+            ),
+            (MaxProcesses, "max-processes", "RLIMIT_NPROC"),
+            (
+                MaxRealtimePriority,
+                "max-realtime-priority",
+                "RLIMIT_RTPRIO",
+            ),
+            (MaxRealtimeTimeout, "max-realtime-timeout", "RLIMIT_RTTIME"),
+            (MaxResidentSet, "max-resident-set", "RLIMIT_RSS"),
+            (MaxStackSize, "max-stack-size", "RLIMIT_STACK"),
+        ];
+
+        for rlimit in rlimits {
+            check_rlimit_strings(rlimit);
+        }
+    }
+}
+
+/// OciDefaultsResourceLimitValue specifies that any u32 integer and "unlimited"
+/// are the valid resource-limits. The hard-limit="unlimited" or
+/// soft-limit="unlimited" sets u64::MAX in spec file for container
+/// that converts to unlimited in containers for that resource
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub enum OciDefaultsResourceLimitValue {
+    Value(u32),
```

Comment:
I am not confident about this - "max file size" seems to take a non-negative i64 value. You should check whether the same range is accepted for other limits.

We have a `NonNegativeInteger` type already (for i32) and could do the same for a `NonNegativeI64` type if needed.

If it turns out that `NonNegativeI64` is the right range for incoming values, then I would prefer to store everything as an i64, and represent "unlimited" as `-1`.